backend/src/main/java/com/google/sps/servlets/FindRecipesServlet.java

```diff
@@ -15,36 +15,59 @@
 package com.google.sps.servlets;
 
 import com.google.gson.Gson;
+import com.google.gson.JsonArray;
+import com.google.gson.JsonElement;
+import com.google.gson.JsonObject;
+import com.google.gson.JsonParser;
 import com.google.sps.data.Recipe;
+import com.google.sps.ApiKeys;
 import com.google.sps.scraping.BBCGoodFoodRecipeScraper;
 import java.io.IOException;
+import java.io.File;
+import org.jsoup.Jsoup;
+import org.jsoup.nodes.Document;
 import java.util.ArrayList;
 import java.util.List;
 import javax.servlet.annotation.WebServlet;
 import javax.servlet.http.HttpServletRequest;
 import javax.servlet.http.HttpServletResponse;
 
 /* Servlet that:
- * in Post request, returns a list of recommended recipes based on the ingredients in the request
- */
+* in Post request returns web links based on the inputted ingredients */
 @WebServlet("/api/find-recipes")
 public class FindRecipesServlet extends AuthenticationServlet {
+  private static final int MAX_NUMBER_OF_RECIPES = 3;
+  private static final String key = ApiKeys.customSearchKey;
+  
   @Override
   protected void get(HttpServletRequest request, HttpServletResponse response) throws IOException {
-    // no get request
+      //no get request
   }
 
-  // TODO: based on the ingredients in the request, retrieve 5 links to scrape recipes from
-  // now: returns a list of recipes scraped from bbc good food
   @Override
   protected void post(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    String ingredients = request.getParameter("ingredients");
```

Comment:
I thought this is how we retrieved ingredients: `String ingredients = request.getReader().readLine();`